libafl/src/monitors/statsd.rs

```diff
@@ -0,0 +1,208 @@
+//! StatsD monitor.
+//!
+//! This roughly corresponds to the [AFL++'s rpc_statsd](https://github.com/AFLplusplus/AFLplusplus/blob/stable/docs/rpc_statsd.md),
+//! so you could view such documentation for detailed information.
+//!
+//! StatsD monitor is useful when you have multiple fuzzing instances, and this monitor
+//! could help visualizing the aggregated fuzzing statistics with serveral third-party
+//! statsd-related tools.
+
+// Use this since clippy thinks we should use `StatsD` instead of StatsD.
+#![allow(clippy::doc_markdown)]
+
+use alloc::string::String;
+use std::{borrow::ToOwned, net::UdpSocket};
+
+use cadence::{BufferedUdpMetricSink, Gauged, QueuingMetricSink, StatsdClient};
+use libafl_bolts::ClientId;
+
+use super::{
+    stats::{manager::GlobalStats, ClientStatsManager, EdgeCoverage, ItemGeometry},
+    Monitor,
+};
+
+/// StatsD monitor
+#[derive(Debug)]
+pub struct StatsdMonitor {
+    target_host: String,
+    target_port: u16,
+    tag_flavor: StatsdMonitorTagFlavor,
+    statsd_client: Option<StatsdClient>,
+}
+
+const METRIC_PREFIX: &str = "fuzzing";
+
+/// Flavor of StatsD tag
+#[derive(Debug)]
+pub enum StatsdMonitorTagFlavor {
+    /// [Datadog](https://docs.datadoghq.com/developers/dogstatsd/) style tag
+    DogStatsd {
+        /// Identifier to distinguish this fuzzing instance with others.
+        tag_identifier: String,
```

Comment:
Cow?

Reply:
This string is never cloned, so a String may be enough?